internal/service/vpc/datasource_project_vpc.go

```diff
@@ -24,24 +32,58 @@ func datasourceProjectVPCRead(_ context.Context, d *schema.ResourceData, m inter
 
 	projectName := d.Get("project").(string)
 	cloudName := d.Get("cloud_name").(string)
+	vpcId, hasId := d.GetOk("id")
 
 	vpcs, err := client.VPCs.List(projectName)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
+	filtered_vpcs := make([]*aiven.VPC, 0)
+
 	for _, vpc := range vpcs {
-		if vpc.CloudName == cloudName {
-			d.SetId(schemautil.BuildResourceID(projectName, vpc.ProjectVPCID))
-			err = copyVPCPropertiesFromAPIResponseToTerraform(d, vpc, projectName)
-			if err != nil {
-				return diag.FromErr(err)
+		condition := vpc.CloudName == cloudName
+
+		// We don't need to care about the cloud name if the ID is provided
+		if hasId {
+			split_id := schemautil.SplitResourceID(vpcId.(string), 2)
+			if len(split_id) < 2 {
+				return diag.Errorf("Invalid VPC id %s", vpcId.(string))
 			}
 
-			return nil
+			id := split_id[1]
+			condition = vpc.ProjectVPCID == id
 		}
+
+		if condition {
+			filtered_vpcs = append(filtered_vpcs, vpc)
+		}
+	}
+
+	if len(filtered_vpcs) == 0 {
+		return diag.Errorf("project %s has no VPC defined for %s",
+			projectName, cloudName)
+	}
+
+	if len(filtered_vpcs) > 1 {
+		// List out the available options in the error message
+		var smg string
+		for _, vpc := range filtered_vpcs {
+			id := schemautil.BuildResourceID(projectName, vpc.ProjectVPCID)
+			smg = smg + fmt.Sprintf("- ID=(%v) , State=(%v) - NetworkCIDR=(%v)\n", id, vpc.State, vpc.NetworkCIDR)
+		}
+		return diag.Errorf("project %s has multiple VPC defined for %s. Please add `id` to get the desired one. The available vpc ids are:\n%s",
+			projectName, cloudName, smg)
+	}
+
+	vpc := filtered_vpcs[0]
+
+	d.SetId(schemautil.BuildResourceID(projectName, vpc.ProjectVPCID))
+
+	err = copyVPCPropertiesFromAPIResponseToTerraform(d, vpc, projectName)
```

Comment:
I think it is best that we call here `resourceProjectVPCRead` instead of `copyVPCPropertiesFromAPIResponseToTerraform`

Reply:
hmm, using `resourceProjectVPCRead` would call the API once again to get the data when we already have it, so I think it's not necessary. :thinking: Also we used `copyVPCPropertiesFromAPIResponseToTerraform` in the old code as well so using it again here is kinda OK I think :sweat_smile: